### ISSUE TITLE
Nether Portals not linking correctly

### ISSUE THREAD

**Issue by Pirgosth** [rlf/uSkyBlock#1260]
_Please paste the output from `/usb version` below_
```[14:00:31] [Server thread/INFO]: Name: uSkyBlock
[14:00:31] [Server thread/INFO]: Version: 2.8.7
[14:00:31] [Server thread/INFO]: Description: Ultimate SkyBlock v2.8.7-73e1fc4
[14:00:31] [Server thread/INFO]: Language: en (en)
[14:00:31] [Server thread/INFO]:   State: d=128, r=64, i=2, p=1, n=true, awe=false
[14:00:31] [Server thread/INFO]: Server: CraftBukkit git-Spigot-2f5d615-cfeef75 (MC: 1.15.2)
[14:00:31] [Server thread/INFO]:   State: online=false, bungee=false
[14:00:31] [Server thread/INFO]: ------------------------------
[14:00:31] [Server thread/INFO]: Vault 1.7.2-b107 (ENABLED)
[14:00:31] [Server thread/INFO]: WorldEdit 7.1.0;8e55131 (ENABLED)
[14:00:31] [Server thread/INFO]: WorldGuard 7.0.2-SNAPSHOT;bf2593e (ENABLED)
[14:00:31] [Server thread/INFO]: Multiverse-Core 4.1.0-b775 (ENABLED)
[14:00:31] [Server thread/INFO]: Multiverse-Portals 4.1.0-b805 (ENABLED)
[14:00:31] [Server thread/INFO]: Multiverse-NetherPortals 4.1.0-b763 (ENABLED)
[14:00:31] [Server thread/INFO]: ------------------------------
```

_What steps will reproduce the problem?_
1.Make a fresh install of a 1.15.2 spigot server
2.Create an island with /is create command
3.Build a nether portal and go to the nether

_If you have any log-files, please paste them to [pastebin.com](http://pastebin.com)_
* server-log: https://pastebin.com/H39ex1fP

Explanation: when entering a nether portal, I have a very strange issue, the nether portal linking is not working properly, and another portal is created on top of the schematic one, and the player need to broke the new one and reactivate the old one to make the link working correctly.

Here are some pictures:

![2020-05-27_14 01 59](https://user-images.githubusercontent.com/22637850/83019039-92c7f200-a026-11ea-8c6b-c0fc094676df.png)
![2020-05-27_14 02 16](https://user-images.githubusercontent.com/22637850/83019043-93f91f00-a026-11ea-8172-effd9ead8bec.png)

**Comment by Pirgosth** [comment]
Am I the only one with this issue ? I've tried several times to create a new spigot server, I re-downloaded all plugin dependancies and uSkyblock itself, using all default configurations and I still do have this problem

**Comment by Muspah** [comment]
There are a few reports of this issue, but I'm not able to reproduce at the moment. We're running a small Dutch server including test builds of USB where the nether works as expected.

**Comment by Pirgosth** [comment]
Yes this is really disturbing, I have a friend running a pretty similar server, and nether portals work just fine for him

**Comment by Pirgosth** [comment]
Could you give me your exact plugin and dependencies versions ? Even if we can't fix this issue for now I'd like to find a workaround to make portals work on my server. I'll really appreciate it.

**Comment by BlackBeltPanda** [comment]
I had this issue a long time ago, so not sure if this fix still works or if yours is caused by something else. What I had to do was disable the nether in server.properties, generate it with Multiverse, and then link it to the skyblock world with Multiverse-Netherportals. I also had to set the scale for the nether world in Multiverse's worlds.yml file to 8.

**Comment by Pirgosth** [comment]
After several hours of testing, I have notice a few things:

- Until you edit the `lastIslandConfig.yml` and change lastIslandX and lastIslandY, the bug appears every time I create or restart a new Island.

- When you have changed `lastIslandConfig.yml`, portal linking works well when CREATING an island for the FIRST TIME, but as far as you restart a new one, the problem appears again.

- The only way I got to make portal linking to work again is to remove the island from uSkyblock configuration (remove player and island folder) to create a new island instead of using the restart command (No need to recreate skyworld_nether world)

@BlackBeltPanda I've tried your fix, but it doesn't change anything, and in my `worlds.yml`, the nether scale was already to 8.

**Comment by Pirgosth** [comment]
@Muspah Sorry about the mention, but have you try restarting your island and then go into the nether ? Because it can seem to work at first when you have just created an island, but as far as you restart a new one the issue is back

**Comment by Muspah** [comment]
It's going to be tomorrow for me, we're having troubles with the incoming power tonight.

**Comment by Pirgosth** [comment]
No problem, hope it will be fix soon =)

**Comment by Pirgosth** [comment]
Ok, because I can't find any way to make this working, and because of another issue with portals (when your portal is to near from another island, it teleport you to the wrong nether island) I'll try to implement a new linking mechanism for uSkyblock, independent of Mutliverse-NetherPortals. If I get this to work, I'll submit a pull request, hope it will bring me somewhere.

**Comment by Muspah** [comment]
The bigger issue is the removal of the TravelAgent in Spigot 1.14.

**Comment by Pirgosth** [comment]
>I've thought about switching to a system where going through a netherportal would just place you at a configured location, something like /is home for the nether, to prevent a lot of issues.

It is exactly what I've been thinking of

**Comment by Muspah** [comment]
Yeah, I've fiddled around with similar implementations around a year ago, but there are some other downsides. Especially when going back from the nether to the overworld when one has goldfarms on the island. Which portal should be used then?

In the end, the easiest solution without a lot of edge-cases seemed teleporting to the island home. Bigger question then, do we want to implement that change for all users, if only a few are experiencing issues? It does break the vanilla expectations from users.

**Comment by Pirgosth** [comment]
Yep, I just realized it would be more difficult than I though. Maybe this can be a option in main config ? Switching between Vanilla or custom but safe behaviour ?

**Comment by Muspah** [comment]
To be honest, I do prefer to switch to a simple implementation if possible. I've just started adding some abstracting between plugins and uSkyBlock, so we're able to add support for different world managers in the future for example.

**Comment by Pirgosth** [comment]
I do understand, maybe should I fork your repos and make my own implementation ?

**Comment by Muspah** [comment]
![image](https://user-images.githubusercontent.com/43544407/83254021-eca2f600-a1a5-11ea-9f27-9b59c0f95c5d.png)
This is what we're running at the moment.

**Comment by Pirgosth** [comment]
Thank you, are you having any issues when restarting an island ?

**Comment by Muspah** [comment]
> Thank you, are you having any issues when restarting an island ?
Nope, no issues

**Comment by Muspah** [comment]
> I do understand, maybe should I fork your repos and make my own implementation ?

Feel free to fork and PR a change, but I would like to keep it opt-in at first, maybe even with a hidden config option that one needs to manually add, to allow us some time to test things out in production.

**Comment by Pirgosth** [comment]
> Nope, no issues

I'm jealous x)

I agree to add a hidden option for test purpose, and to let you the time to get sure everything's right

**Comment by Muspah** [comment]
Mhm, I don't know if that's something to be jealous of, we're running local test builds with broken SQL support, et cetera ;p

**Comment by Pirgosth** [comment]
The pleasure of being a developer :p

**Comment by Muspah** [comment]
Oh no, no, that's fiddling with old Delphi crap, still trying to keep it alive and running after 20 years. A lot of code is in Dutch too 😂

**Comment by Pirgosth** [comment]
:joy:

**Comment by Pirgosth** [comment]
Okay, I'm glad to say that I just finished my new linking mechanism, and it just work perfectly for now!
Unfortunately, I based my work on an old commit (73e1fc4ac55361632b6ab8b05050229c14a187d5, the 2.8.7 release one) and I made a lot of small changes in several files, moreover I added some data in players config files, and I don't think it would be a good idea to add a toggle for this in the main config, even a hidden one, because it will require a lot of work for not much ..

**Comment by Muspah** [comment]
> and I made a lot of small changes in several files, moreover I added some data in players config files, and I don't think it would be a good idea to add a toggle for this in the main config, even a hidden one, because it will require a lot of work for not much ..

It's not a problem, just open a PR when you're ready to. We can always merge to a new branch and do some testing before pushing to master.

**Comment by Pirgosth** [comment]
Alright, I'll send a PR when everything is done ;)
